src/main/java/org/cef/CefApp.java

```diff
@@ -23,605 +25,564 @@
-   * This function may be called multiple times to change or remove the factory
-   * that matches the specified |scheme_name| and optional |domain_name|.
-   * Returns false if an error occurs. This function may be called on any thread
-   * in the browser process.
-   */
-  public boolean registerSchemeHandlerFactory(String schemeName,
-                                              String domainName,
-                                              CefSchemeHandlerFactory factory) {
-    try {
-      return N_RegisterSchemeHandlerFactory(schemeName, domainName, factory);
-    } catch (Exception err) {
-      err.printStackTrace();
+    /**
+     * To shutdown the system, it's important to call the dispose
+     * method. Calling this method closes all client instances with
+     * and all browser instances each client owns. After that the
+     * message loop is terminated and CEF is shutdown.
+     */
+    public synchronized final void dispose() {
+        switch (getState()) {
+            case NEW:
+                // Nothing to do inspite of invalidating the state
+                setState(CefAppState.TERMINATED);
+                break;
+
+            case INITIALIZING:
+            case INITIALIZED:
+                // (3) Shutdown sequence. Close all clients and continue.
+                setState(CefAppState.SHUTTING_DOWN);
+                if (clients_.isEmpty()) {
+                    shutdown();
+                } else {
+                    // shutdown() will be called from clientWasDisposed() when the last
+                    // client is gone.
+                    for (CefClient c : clients_) {
+                        c.dispose();
+                    }
+                }
+                break;
+
+            case NONE:
+            case SHUTTING_DOWN:
+            case TERMINATED:
+                // Ignore shutdown, CefApp is already terminated, in shutdown progress
+                // or was never created (shouldn't be possible)
+                break;
+        }
     }
-    return false;
-  }
-
-  /**
-   * Clear all registered scheme handler factories. Returns false on error. This
-   * function may be called on any thread in the browser process.
-   */
-  public boolean clearSchemeHandlerFactories() {
-    try {
-      return N_ClearSchemeHandlerFactories();
-    } catch (Exception err) {
-      err.printStackTrace();
+
+    /**
+     * Creates a new client instance and returns it to the caller.
+     * One client instance is responsible for one to many browser
+     * instances
+     * @return a new client instance
+     */
+    public synchronized CefClient createClient() {
+        switch (getState()) {
+            case NEW:
+                setState(CefAppState.INITIALIZING);
+
+                //montoyo: Modified for MCEF
+                /*
+                  context.setDaemon(true);  // from older JCEF?
+                  context.start();          // from older JCEF?
```

Comment:
we should probably remove all this dead code because newer JCEF doesn't have it at all